storage/txcache/txCacheFailsafe.go

```diff
@@ -0,0 +1,140 @@
+package txcache
+
+import (
+	"github.com/ElrondNetwork/elrond-go/core"
+	"github.com/ElrondNetwork/elrond-go/storage"
+)
+
+var _ storage.Cacher = (*txCacheFailsafeDecorator)(nil)
+var _ txCache = (*txCacheFailsafeDecorator)(nil)
+
+// txCacheFailsafeDecorator is a decorator over a cache, a decorator which wraps some operations in try / catch constructs
+type txCacheFailsafeDecorator struct {
+	backingCache txCache
+}
+
+func newTxCacheFailsafe(cache txCache) txCache {
+	return &txCacheFailsafeDecorator{
+		backingCache: cache,
+	}
+}
+
+// AddTx delegates to backing cache, with try / catch
+func (decorator *txCacheFailsafeDecorator) AddTx(tx *WrappedTransaction) (ok bool, added bool) {
+	core.TryCatch(func() {
+		ok, added = decorator.backingCache.AddTx(tx)
```

Comment:
Why is this allowed to panic?

Reply:
Removed failsafe decorator. 

Code not allowed to panic, was just an extra safety measure in case of hidden bugs activated by unknown edge cases.